src/convertToObject.js

```diff
@@ -5,8 +5,59 @@
  *
  * @return {object}
  */
+
+const complexStylesString = `
+  background-color:      #fff;
+-webkit-border-radius: 5px;
+  border-radius     : 5px;
+  border: 1px solid #e8e8e8;
+  -webkit-box-sizing: border-box;
+  box-sizing: border-box;
+  clear   : both       ;
+  cursor: pointer;
+  float: left;
+  font-family: inherit;
+      font-size: 14px;
+  font-weight: 400;
+  height: 42px;
+  line-height:    40px;
+  outline: 0;
+  padding-left    : 18px;
+  padding-right: 30px;
+  ;
+
+  ;
+  position: relative;
+
+
+  text-align: left !important;
+  -webkit-transition: all .2s ease-in-out;
+  transition: all .2s ease-in-out;
+  -moz-user-select: none;
+  -ms-user-select: none;
+  user-select: none;
+
+
+  white-space: nowrap;
+  width: auto;
+`;
+
 function convertToObject(sourceString) {
-  // write your code here
+  const splittedStyles = sourceString.split(';');
+  const formattedObject = {};
+
+  for (const splittedStyle of splittedStyles) {
```

Comment:
[CODE PRACTICE] - It's recommended to use array iteration methods instead of loops. For example, you could use `forEach` or `reduce` to iterate over `splittedStyles` and build `formattedObject`.